lib/Mojo/Feed/Item.pm

```diff
@@ -1,125 +1,107 @@
 package Mojo::Feed::Item;
 use Mojo::Base '-base';
-has [qw(title link content id description guid published author _raw)];
+use Mojo::Util 'monkey_patch';
+use Mojo::Feed::Item::Enclosure;
+use HTTP::Date 'str2time';
+has [qw(title link content id description guid published author)];
 
 has tags => sub {
-  $item->find('category, dc\:subject')
-    ->each(sub { $_[0]->text || $_[0]->attr('term') });
+  shift->dom->find('category, dc\:subject')
+    ->map(sub { $_[0]->text || $_[0]->attr('term') });
 };
 
 has 'dom';
 
 has summary => sub { shift->description };
 
-our @time_fields
-  = (qw(pubDate published created issued updated modified dc\:date));
-our %is_time_field = map { $_ => 1 } @time_fields;
-
-
--  foreach my $k (
--    qw(title id summary guid content description content\:encoded xhtml\:body dc\:creator author),
--    @time_fields
--    )
--  {
--    my $p = $item->at($k);
--    if ($p) {
--
--      # skip namespaced items - like itunes:summary - unless explicitly
--      # searched:
--      next
--        if ($p->tag =~ /\:/
--        && $k ne 'content\:encoded'
--        && $k ne 'xhtml\:body'
--        && $k ne 'dc\:date'
--        && $k ne 'dc\:creator');
--      $h{$k} = $p->text || $p->content;
--      if ($k eq 'author' && $p->at('name')) {
--        $h{$k} = $p->at('name')->text;
--      }
--      if ($is_time_field{$k}) {
--        $h{$k} = str2time($h{$k});
--      }
--    }
--  }
+my %alternatives = (
+  content     => ['content\:encoded', 'xhtml\:body', 'description'],
+  description => ['summary'],
+  published =>
+    ['pubDate', 'dc\:date', 'created', 'issued', 'updated', 'modified'],
+  author => ['dc\:creator'],
+  id     => ['guid', 'link'],
+);
+
+monkey_patch 'Mojo::DOM' => at_with_namespace => sub {
+  my ($dom, $field) = @_;
+  if (my $p = $dom->at($field)) {
+    my $tag = $p->tag;
+    $tag =~ s/:/\\:/;
+    return $p if $tag eq $field;
+  }
+  return;
+};
 
-has enclosures => sub {
+foreach my $k (qw(title link content id description guid published author)) {
+  has $k => sub {
     my $self = shift;
-    my @enclosures;
-    $self->dom->find('enclosure')->each(
-        sub {
-            push @enclosures, shift->attr;
-        }
-    );
-    $self->dom->find('link')->each(
-        sub {
-            my $l = shift;
-            if (   $l->attr('href')
-                && $l->attr('rel')
-                && $l->attr('rel') eq 'enclosure' )
-            {
-                push @enclosures,
-                  {
-                    url    => $l->attr('href'),
-                    type   => $l->attr('type'),
-                    length => $l->attr('length')
-                  };
-            }
-        }
-    );
-    return Mojo::Collection->new(
-        map { Mojo::Feed::Enclosure->new($_) } @enclosures;
-    );
+    my @alternatives = @{$alternatives{$k} || []};
+
+    my $p;
+    for my $field ($k, @alternatives) {
+      $p = $self->dom->at_with_namespace($field);
```

Comment:
What was this trying to solve? Is there a test failure that made you add this, and if not, could you add one?

Reply:
Sorry, i'm not sure what do you mean with _this_? This is a pretty big diff over a lot of commits...